extensions/agroal/spi/src/main/java/io/quarkus/agroal/deployment/JdbcDriverBuildItem.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.agroal.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Register a JDBC driver for the Agroal extension.
+ * <p>
+ * It allows to resolve automatically the driver from the kind, thus users don't have to set the driver anymore, except if they
+ * want to use a specific one.
+ */
+public final class JdbcDriverBuildItem extends MultiBuildItem {
+
+    private final String kind;
+
+    private final String driverClass;
+
+    private final Optional<String> xaDriverClass;
+
+    public JdbcDriverBuildItem(String kind, String driverClass, Optional<String> xaDriverClass) {
```

Comment:
I can't help but think that the last parameter can be sent as null by other extensions. In order to prevent that, I'd suggest it to be String too and in the constructor body just use an `Optional.ofNullable`

Reply:
I know it's not a pattern people love but I want to be clear it's not mandatory.

Reply:
In the end I decided to expose two different constructors.